### PR TITLE
Rewards summary message on game end

### LINKED ISSUE #206: End screen- Reward Summary
Please add this reward summary to bedwars1058's game ending 
https://hypixel.net/attachments/1600890202852-png.2006850/

### PR DESCRIPTION
**Pull request by reussy**
This implements the summary message and closes #206

 **Some methods are moved by Intellij reformat code tool!**
- New maps for keep the current coins and experience earned until game end. It's unnecessary create classes for this (I guess).
- These maps are loaded on player join and destroyed on player leave.
- These maps are updated on `addXp` method and `MoneyListeners` methods.
- The message is implemented below of top killers message.
- The radical implements of "long progress bar" is for use in chat. The shorter progress bar it looks very ugly in the rewards summary message. **This could be improved or customized from level.yml in the future.**

Everything was tested by me. It can be tested by you at `mc.exoduscraft.club - BedWars Multiarena`
Looks like this: https://i.imgur.com/rDPjXkw.png

### CODE REVIEW

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/levels/internal/PlayerLevel.java [hunk 1]
```diff
@@ -33,21 +34,19 @@
 @SuppressWarnings("WeakerAccess")
 public class PlayerLevel {
 
-    private UUID uuid;
+    private static final ConcurrentHashMap<UUID, PlayerLevel> levelByPlayer = new ConcurrentHashMap<>();
```
Comment (andrei1058) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/levels/internal/PlayerLevel.java:37]: why is this concurrent?

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/arena/Arena.java [hunk 2]
```diff
@@ -158,6 +159,9 @@
     private HashMap<Player, Integer> playerDeaths = new HashMap<>();
     private HashMap<Player, Integer> playerFinalKillDeaths = new HashMap<>();
 
+    /* SUMMARY REWARDS */
+    private static final Map<UUID, Integer> coinsEarned = new HashMap<>();
```
Comment (andrei1058) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/arena/Arena.java:163]: it should not be static

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/levels/internal/PlayerLevel.java [hunk 3]
```diff
@@ -174,8 +210,12 @@ public void addXp(int xp, PlayerXpGainEvent.XpSource source) {
         this.currentXp += xp;
         upgradeLevel();
         updateProgressBar();
+        updateLongProgressBar();
         Bukkit.getPluginManager().callEvent(new PlayerXpGainEvent(Bukkit.getPlayer(uuid), xp, source));
         modified = true;
+        if (Arena.getExperienceEarned().containsKey(uuid)) {
+            Arena.getExperienceEarned().put(uuid, Arena.getExperienceEarned(uuid) + xp);
```
Comment (andrei1058) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/levels/internal/PlayerLevel.java]: this must be handled with an event listener XpGainEvent etc.

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/money/internal/MoneyPerMinuteTask.java [hunk 4]
```diff
@@ -26,6 +26,7 @@ public MoneyPerMinuteTask(Arena arena) {
             for (Player p : arena.getPlayers()) {
                     BedWars.getEconomy ().giveMoney ( p, money );
                     p.sendMessage ( Language.getMsg ( p, Messages.MONEY_REWARD_PER_MINUTE ).replace ( "{money}", String.valueOf ( money ) ) );
+                    Arena.getCoinsEarned().put(p.getUniqueId(), Arena.getCoinsEarned(p.getUniqueId()) + money);
```
Comment (andrei1058) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/money/internal/MoneyPerMinuteTask.java]: we should implement a new event like BedWarsMoneyGainEvent and listen to that in otder to track coins earned

**Review by andrei1058 — CHANGES_REQUESTED** [review]

**Comment by andrei1058** [comment]
bump

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/levels/internal/PlayerLevel.java (hunk 1 above)
Comment (reussy) [reply bedwars-plugin/src/main/java/com/andrei1058/bedwars/levels/internal/PlayerLevel.java:37]: I think that's was touched by "reformat code" tool by Intellij. In the current code it is concurrent. I must have missed it when sending the pr

image

**Comment by ghost** [comment]
When will this be implemented

**Comment by reussy** [comment]
Well, I don't know what the hell happened but well, I'd better make a new pr :/

**Pull request closed without merging**